.circleci/config.yml

```diff
@@ -1,29 +1,54 @@
 workflows:
   version: 2
   circleci_tests:
+    # The following jobs runs in the following stages:
+    # - Setup runs first.
+    # - After setup passes, lint and frontend test run.
+    #     - If frontend tests fail, stop and do not run the next step, but lint continues.
+    # - After frontend tests passes, backend and typescript tests run.
+    #     - If typescript tests fail, stop and do not run the next step, but backend continues.
+    # - After typescript tests passes, the flagship e2e tests (navigation and file upload) run.
```

Comment:
Add a couple of sentences: We run these tests as sanity checks before running the entire test suite. Note that the navigation e2e test also compiles all the prod files needed for other prod e2e tests.

Also add: if either of these tests fail, the rest of the e2e tests in the corresponding suite do not run (or similar).

Reply:
done!